src/metabase/sync/analyze/fingerprint.clj

```diff
@@ -160,8 +160,8 @@
     [:not (mdb.query/isa :semantic_type :type/PK)]
     [:= :semantic_type nil]]
    [:not-in :visibility_type ["retired" "sensitive"]]
-   [:not= :base_type (u/qualified-name :type/*)]
-   [:not (mdb.query/isa :base_type :type/Structured)]])
+   [:not-in :base_type (-> (set (mapcat mdb.query/type-keyword->descendants [:type/Structured :type/Collection]))
+                           (conj (u/qualified-name :type/*)))]])
```

Comment:
It feels a bit low level to deal directly with descendant sets here, what's wrong with just using `isa` twice? If you're set on this approach I would at leave use a `reduce set/union` to consolidate the sets a bit more directly.

Reply:
Ah, it looks like big motivator was to support nice tests?

Reply:
> motivator was to support nice tests?

Yes. I didn't think it was a big deal to use `mdb.query/type-keyword->descendants` here because `mdb.query/isa` doesn't hide any information that is likely to change. I see it more of a convenience function than actually providing useful indirection. The only thing it really hides is the usage of `:in`, but set inclusion is guaranteed to feature in any implementation of the `isa` function, and seems unlikely to change.

Reply:
> use a `reduce set/union`

This is subjective of course but using `mapcat` is slightly shorter, so I prefer it

Reply:
Subjectivity strikes again. What I don't like about `descendants` is that it assumes things are enumerable and already loaded, it also calculates those set on the fly. In practice none of these a problems, and I do like how it makes the tests super explicit 👍